xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -63,6 +72,48 @@ bool IsBatchedMatmul(const HloInstruction* instr) {
   return is_batch_dot;
 }
 
+// We need to check if current gemm is sharing a parent node with a forward
+// fMHA call. We check this by doing a BFS of all operands to see if there's
+// any user that is a forward fMHA custom call.
```

Comment:
Is it possible to motivate the reader why do we need to find out whether it's shared?

Reply:
changed doc to explain this